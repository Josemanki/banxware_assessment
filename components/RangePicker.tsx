import React, { Dispatch, SetStateAction } from 'react';
import { TDateRange } from '../types';

export type Props = {
  possibleRanges: TDateRange[];
  range: TDateRange;
  setRange: Dispatch<SetStateAction<TDateRange>>;
};

const RangePicker = ({ possibleRanges, range, setRange }: Props) => {
  return (
    <div className="range-picker">
      {possibleRanges.map((possibleRange, index) => (
        <button
          key={index}
          className={`range-picker__range${range === possibleRange ? ' selected' : ''}`}
          onClick={() => {
            setRange(possibleRange);
          }}
        >
          {possibleRange}
        </button>
      ))}
    </div>
  );
};

export default RangePicker;
