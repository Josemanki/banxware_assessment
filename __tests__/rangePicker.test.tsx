import { render, fireEvent } from '@testing-library/react';
import RangePicker, { Props } from '../components/RangePicker';
import '@testing-library/jest-dom';

jest.mock('../components/Chart');

const setRange = jest.fn();
const mockProps: Props = { possibleRanges: ['1D', '5D', '1M', '3M', '6M'], range: '3M', setRange: setRange };

test('Should render the rangePicker component', () => {
  const { asFragment } = render(<RangePicker {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Should render 5 button components', () => {
  const { queryAllByRole } = render(<RangePicker {...mockProps} />);
  expect(queryAllByRole('button')).toHaveLength(5);
});

test('Should call setRange function when button clicked', () => {
  const { getByText } = render(<RangePicker {...mockProps} />);

  const sixMonthsButton = getByText('6M');

  fireEvent.click(sixMonthsButton);
  expect(setRange).toHaveBeenCalled();
});
