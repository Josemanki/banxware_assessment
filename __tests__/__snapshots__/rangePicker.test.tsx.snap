// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the rangePicker component 1`] = `
<DocumentFragment>
  <div
    class="range-picker"
  >
    <button
      class="range-picker__range"
    >
      1D
    </button>
    <button
      class="range-picker__range"
    >
      5D
    </button>
    <button
      class="range-picker__range"
    >
      1M
    </button>
    <button
      class="range-picker__range selected"
    >
      3M
    </button>
    <button
      class="range-picker__range"
    >
      6M
    </button>
  </div>
</DocumentFragment>
`;
